pkg/operator/vspherecontroller/checks/check_error.go

```diff
@@ -163,11 +191,74 @@ func CheckClusterStatus(result ClusterCheckResult, apiDependencies KubeAPIInterf
 
 	case CheckActionBlockUpgrade:
 		return ClusterCheckBlockUpgrade, result
-
+	case CheckActionRequiresAdminAck:
+		return checkForIntreePluginUse(result, apiDependencies)
 	case CheckActionBlockUpgradeDriverInstall:
 		return ClusterCheckBlockUpgradeDriverInstall, result
-
 	default:
 		return ClusterCheckAllGood, result
 	}
 }
+
+// returns false is migration is enabled in the cluster.
+// returns true if migration is not enabled in the cluster and cluster is using
+// in-tree vSphere volumes.
+func checkForIntreePluginUse(result ClusterCheckResult, apiDependencies KubeAPIInterface) (ClusterCheckStatus, ClusterCheckResult) {
+	storageCR, err := apiDependencies.GetStorage(storageOperatorName)
+	if err != nil {
+		reason := fmt.Errorf("vsphere csi driver installed failed with %s, unable to verify storage status: %v", result.Reason, err)
+		return ClusterCheckDegrade, MakeClusterDegradedError(CheckStatusOpenshiftAPIError, reason)
+	}
+
+	klog.Infof("Checking for intree plugin use")
+
+	allGoodCheckResult := MakeClusterCheckResultPass()
+
+	// migration is enabled and hence we should be fine
+	driverName := storageCR.Spec.VSphereStorageDriver
+	if driverName == v1.CSIWithMigrationDriver {
+		return ClusterCheckAllGood, allGoodCheckResult
+	}
+
+	pvs, err := apiDependencies.ListPersistentVolumes()
+	if err != nil {
+		reason := fmt.Errorf("vsphere csi driver installed failed with %s, unable to list pvs: %v", result.Reason, err)
+		return ClusterCheckDegrade, MakeClusterDegradedError(CheckStatusOpenshiftAPIError, reason)
+	}
+
+	usingvSphereVolumes := false
+
+	for _, pv := range pvs {
+		if pv.Spec.VsphereVolume != nil {
+			usingvSphereVolumes = true
```

Comment:
Would it be useful to log what PV has in-tree volume? In case an user argues they moved all PVs to CSI.

Reply:
added logs

Reply:
Err.. I meant log the PV names.

Reply:
hmm, I wonder if that will be a bit much. In larger clusters this could be a big string...